applications/dashboard/controllers/class.entrycontroller.php

```diff
@@ -1106,6 +1110,7 @@ public function signIn($method = false, $arg1 = false) {
                                 Gdn::userModel()->setField(val('UserID', $user), ['Password' => $pw, 'HashMethod' => 'Vanilla']);
                             }
 
+                            Gdn::userModel()->fireEvent('BeforeSignIn', ['UserID' => $user->UserID]);
```

Comment:
Keeping this consistent with `Gdn_Session` is a little muddier, since we want to avoid using `val`. The next best thing is probably null coalescence.

```suggestion
                            Gdn::userModel()->fireEvent('BeforeSignIn', ['UserID' => $user->UserID ?? false]);
```